### ISSUE TITLE
Possible inconsistencies of origin and direction values of _SpectroscopicObserver0DBase 

### ISSUE THREAD

**Issue by Mateasek** [cherab/core#322]
The [_SpectroscopicObserver0DBase](https://github.com/cherab/core/blob/dc74ba308c1e7a6ac179530ca8092b2e0d9f2e88/cherab/tools/observers/spectroscopy/base.py#L26) uses [origin](https://github.com/cherab/core/blob/dc74ba308c1e7a6ac179530ca8092b2e0d9f2e88/cherab/tools/observers/spectroscopy/base.py#L49) and [direction](https://github.com/cherab/core/blob/dc74ba308c1e7a6ac179530ca8092b2e0d9f2e88/cherab/tools/observers/spectroscopy/base.py#L66) properties to calculate the transform of an observer and keeps the values. This can lead to inconsistencies between the actual transform of the observer and saved values of origin and direction, when the transform is changed directly.

Instead of keeping the values in _origin and _direction I propose changing the property getters to calculate the value from the transform:

``` python
    @property
    def direction(self):
        # The observation direction of the sight line.
        return Vector3D(0, 0, 1).transform(self.transform)

        @property
    def origin(self):
        # The origin point of the sight line.
        return Point3D(0, 0, 0).transform(self.transform)
```
I will be happy to apply this, but since it is work of @vsnever I would like to hear is opinion

**Comment by Mateasek** [comment]
At the end I needed the fix now for my work, @vsnever please feel free to accept/reject it.

**Comment by vsnever** [comment]
Fixed in #323.
